code/modules/client/preference_setup/global/01_ui.dm

```diff
@@ -125,6 +129,20 @@
 			pref.ooccolor = new_ooccolor
 			return TOPIC_REFRESH
 
+	else if ("select_fps")
+		var/version_message
+		if (user.client && user.client.byond_version < 511)
+			version_message = "\nYou need to be using byond version 511 or later to take advantage of this feature, your version of [user.client.byond_version] is too low"
+		if (world.byond_version < 511)
+			version_message += "\nThis server does not currently support client side fps. You can set now for when it does."
+		var/desiredfps = input(user, "Choose your desired fps.[version_message]\n(0 = synced with server tick rate (currently:[world.fps]))", "Character Preference")  as null|num
+		if (!isnull(desiredfps))
+			pref.clientfps = desiredfps
```

Comment:
```suggestion
			pref.clientfps = sanitize_integer(desiredfps, 0, 1000, initial(pref.clientfps))
```

gotta sanitize user input. Otherwise they'll shove a negative number here and RIP goes everything.

Reply:
can't you use `Clamp()`?

Reply:
`sanitize_integer` acts a bit differently, it has a default value, instead of replacing with an extremum.